service-base/src/main/java/org/eclipse/hono/service/registration/RegistrationService.java

```diff
@@ -117,40 +78,22 @@
-     * @param tenantId The tenant the device belongs to.
-     * @param deviceId The ID of the device to update the registration for.
-     * @param otherKeys A map containing additional properties to be registered with the device (may be {@code null}).
-     *                  The properties provided in this map will completely replace any existing properties registered for the device.
-     * @param resultHandler The handler to invoke with the result of the operation.
-     *             The <em>status</em> will be
-     *             <ul>
-     *             <li><em>204 No Content</em> if the registration information has been updated successfully.</li>
-     *             <li><em>404 Not Found</em> if no device with the given identifier is
-     *             registered for the tenant.</li>
-     *             </ul>
-     * @throws NullPointerException if any of tenant, device ID or result handler is {@code null}.
-     * @see <a href="https://www.eclipse.org/hono/api/device-registration-api/#update-device-registration">
-     *      Device Registration API - Update Device Registration</a>
-     */
-    void updateDevice(String tenantId, String deviceId, JsonObject otherKeys, Handler<AsyncResult<RegistrationResult>> resultHandler);
-
-    /**
-     * Removes a device.
-     * 
+     * Gets device registration data by device ID.
+     *
      * @param tenantId The tenant the device belongs to.
-     * @param deviceId The ID of the device to remove.
+     * @param deviceId The ID of the device to get registration data for.
      * @param resultHandler The handler to invoke with the result of the operation.
      *             The <em>status</em> will be
      *             <ul>
-     *             <li><em>204 No Content</em> if the device has been removed successfully.</li>
-     *             <li><em>404 Not Found</em> if no device with the given identifier is
-     *             registered for the tenant.</li>
+     *             <li><em>200 OK</em> if a device with the given ID is registered
+     *             for the tenant. The <em>payload</em> will contain the properties
+     *             registered for the device.</li>
+     *             <li><em>404 Not Found</em> if no device with the given identifier
+     *             is registered for the tenant.</li>
      *             </ul>
      * @throws NullPointerException if any of the parameters is {@code null}.
-     * @see <a href="https://www.eclipse.org/hono/api/device-registration-api/#deregister-device">
-     *      Device Registration API - Deregister Device</a>
+     * @see <a href="https://www.eclipse.org/hono/api/device-registration-api/#get-registration-information">
+     *      Device Registration API - Get Registration Information</a>
      */
-    void removeDevice(String tenantId, String deviceId, Handler<AsyncResult<RegistrationResult>> resultHandler);
+    void getDevice(String tenantId, String deviceId, Handler<AsyncResult<RegistrationResult>> resultHandler);
```

Comment:
this operation is defined as optional in the API. Is there a particular reason why you want to keep it in this interface?

Reply:
it is used in the assertion process to retrieve the device and verify that a device is properly asserted. https://github.com/eclipse/hono/blob/fb942feacf74171d8b7507782bcbf3d4c1726c90/service-base/src/main/java/org/eclipse/hono/service/registration/BaseRegistrationService.java#L311

We can put the method in the optional interface, but as it would be implemented anyway by the base class. In that case the Complete base class would respect the complete interface only because it extends the base class. (or duplicate code).
I don't know which case is better ?

Reply:
FMPOV it belongs in the `CompleteRegistrationService` interface. We then simply remove the `@Override` annotation from *getDevice* in `BaseRegistrationService` and we are done, right?
I don't think it's a problem that CompleteBaseRegistrationService requires extension of BaseRegistrationService in order to fulfill its contract because we have designed it to an extension of it in the first place. It already needs to extend BaseRegistrationService in order to implement *assertRegistration* as well ...

Reply:
Sounds logical to me !